src/Pim/Component/Catalog/Validator/UniqueValuesSet.php

```diff
@@ -43,27 +43,23 @@ public function reset()
      */
     public function addValue(ProductValueInterface $productValue)
     {
-        // TODO: To be reworked in TIP-698
-        return true;
-
-        $product = $productValue->getProduct();
-        $productIdentifier = $this->getProductIdentifier($product);
-        $productValueData = $this->getValueData($productValue);
-        $uniqueValueCode = $this->getUniqueValueCode($productValue);
+        $identifier = spl_object_hash($productValue);
+        $data = (string) $productValue;
```

Comment:
in other files, you used `$productValue->__toString()`, both are ok to me, but it's to be consistent :)
(I prefer cast, but it's a taste matter :water_buffalo: )